src/data/datasets/representation/validation.py

```diff
@@ -0,0 +1,237 @@
+        self._imgid2paths = {img_id: self._imgid2paths[img_id] for img_id in self.__index2imgid.values()}
+        self._data_len = self.__n_queries + self.__n_relevant
+
+        self.__use_gallery = use_gallery
+
+        if self.__use_gallery:
+            self.__gallery_folder = gallery_folder
+            self.__gallery_list_csv_path = gallery_list_csv_path
+
+            if self.__gallery_folder is None:
+                raise ValueError('Argument `gallery_folder` is None, please send path to gallery_folder')
+            if self.__gallery_list_csv_path is None:
+                raise ValueError('Argument `gallery_list_csv_path` is None, please send path to gallery_list_csv_path')
+
+            self.__gallery_paths = pd.read_csv(Path(self.__gallery_folder) / self.__gallery_list_csv_path,
+                                               usecols=['id', self._input_column],
+                                               dtype={'id': int, self._input_column: str})
+            self.__gallery_imgid2paths = dict(zip(self.__gallery_paths['id'], self.__gallery_paths[self._input_column]))
+
+            self.__n_gallery = 0
+            self._gallery_index2imgid = {}
+            for img_id in self.__gallery_imgid2paths:
+                self._gallery_index2imgid[self._data_len + self.__n_gallery] = img_id
+                self.__n_gallery += 1
+
+            self._data_len += self.__n_gallery
+
+        self.__imgid2index = {}
+        for index, img_id in self.__index2imgid.items():
+            if img_id not in self.__imgid2index:
+                self.__imgid2index[img_id] = index
+
+        self.__scores, self.__is_query = self.__get_targets()
+
+    def __getitem__(self, index: int) -> dict:
+
+        if index < self.__n_queries + self.__n_relevant:
+            img_id = self.__index2imgid[index]
+            image_path = self._data_folder / self._imgid2paths[img_id]
+        else:
+            img_id = self._gallery_index2imgid[index]
+            image_path = self.__gallery_folder / self.__gallery_imgid2paths[img_id]
+
+        image = self._read_image(image_path)
+        sample = {'image': image}
+        sample = self._apply_transform(self.augment, sample)
+        sample = self._apply_transform(self.transform, sample)
+        sample['image'] = sample['image'].type(torch.__dict__[self._input_dtype])
+        sample['index'] = index
+        sample['is_query'] = self.__is_query[index]
+        sample['scores'] = self.__scores[index]
+
+        return sample
+
+    def __parse_match_csv(self):
+        self.__query_arr = self.__matches.loc[:, 'query'].tolist()
+        self.__index2imgid = dict(enumerate(self.__query_arr))
+        self.__n_queries = len(self.__index2imgid)
+
+        self.__relevant_arr, self.__relevance_scores = [], []
+        self.__n_relevant = 0
+
+        for index in range(len(self.__matches)):
+            self.__relevant_arr.append([])
+            self.__relevance_scores.append([])
```

Comment:
We need to define for our team, what is the best way to append new generated list in another base list. I see that constructor first time, and for me it was unclear to read. But if Vlad approve it ok.
Usually i do:
`base_list = []
 for ... :
     curr_added_list = generate_list()
     base_list.append(curr_added_list)
`